scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/EquationSolvingAlgorithms.java

```diff
@@ -301,7 +303,18 @@ public boolean isApproximation() {
         }
       };
 
-      double ff_x_nminus1 = ff.valueAt(x_nminus1), ff_x_n = ff.valueAt(x_n);
+      var ff_x_nminus1 = ff.valueAt(x_nminus1);
+      if (ff_x_nminus1 >= - toleranceYLow &&
+          ff_x_nminus1 <= toleranceYHigh
+          && (x_nminus1 >= dxLow && x_nminus1 <= dxHigh)) {
+        return new RootFindingResult<>(x_nminus1, ff_x_nminus1 + y, new History<>(history));
+      }
+      var ff_x_n = ff.valueAt(x_n);
+      if (ff_x_n >= - toleranceYLow &&
+          ff_x_n <= toleranceYHigh
+          && (x_n >= dxLow && x_n <= dxHigh)) {
+        return new RootFindingResult<>(x_n, ff_x_n + y, new History<>(history));
+      }
       for (int nbIt = 0; nbIt < maxNbIterations; nbIt++) {
         final var localDerivative = (ff_x_n - ff_x_nminus1) / (x_n - x_nminus1);
         if (localDerivative == 0) {
```

Comment:
🔶 Is there a reason we do a `NaN` check in the `Duration` version, but not in the `double` version?

Reply:
There was none, I added it to the double version.